src/Invitation/InvitationForm.js

```diff
@@ -1,16 +1,120 @@
-import React from 'react';
+import React, { useState, useRef } from 'react';
 import { Link } from 'react-router-dom';
-import { Button } from 'react-bootstrap';
-
+import {
+  Container,
+  Card,
+  CardGroup,
+  Form,
+  Stack,
+  Carousel,
+} from 'react-bootstrap';
+import PlumButton from '../Common/styled/buttonstyles/PlumButton';
+import PlumFilledButton from '../Common/styled/buttonstyles/PlumFilledButton';
 import 'bootstrap/dist/css/bootstrap.min.css';
 
 export default function InvitationForm() {
+  const [mainPhoto, setMainPhoto] = useState(
+    'http://tineye.com/images/widgets/mona.jpg'
+  );
+  const [inviteMessage, setInviteMessage] = useState('');
+  const [galleryURL, setGalleryURL] = useState([
+    'http://tineye.com/images/widgets/mona.jpg',
+    'http://tineye.com/images/widgets/mona.jpg',
+    'http://tineye.com/images/widgets/mona.jpg',
+  ]);
+  const galleryArray = [];
+  const hiddenMainPhotoInput = useRef(null);
+  const hiddenGalleryInput = useRef(null);
+
+  const uploadMainPhoto = () => {
+    hiddenMainPhotoInput.current.click();
+  };
+  const addPicToGallery = () => {
+    hiddenGalleryInput.current.click();
+  };
+
+  const previewMainPhoto = (e) => {
+    const reader = new FileReader();
+    reader.onload = () => {
+      if (reader.readyState === 2) {
+        setMainPhoto(reader.result);
+      }
+    };
+    reader.readAsDataURL(e.target.files[0]);
+  };
+
+  const updateMessage = (e) => {
+    setInviteMessage(e.target.value);
+  };
+
+  const previewGallery = (e) => {
+    const selectedPhotos = e.target.files;
+    for (let i = 0; i < selectedPhotos.length; i++) {
+      galleryArray.push(URL.createObjectURL(selectedPhotos[i]));
+    }
+    setGalleryURL([...galleryArray, ...galleryURL]);
+  };
+
   return (
-    <>
-      <h2>This is Invitaion Page</h2>
-      <Link to="/rsvp-preview">
-        <Button>Click me to Go to RSVP Page</Button>
-      </Link>
-    </>
+    <Container
+      style={{
+        width: '65%',
+        // backgroundImage: 'url(http://tineye.com/images/widgets/mona.jpg)',
+        // backgroundRepeat: 'no-repeat',
+        // height: '100%', ----------TO BE REVISED--------------
+      }}
+    >
+      <h2>Your Wedding Website</h2>
+      <Stack className="mb-5" direction="horizontal" gap={3}>
+        <PlumButton className="ms-auto">Share your website</PlumButton>
+        <Link to="/rsvp-preview" state={{ from: galleryURL }}>
+          <PlumFilledButton>Preview Site</PlumFilledButton>
+        </Link>
+      </Stack>
+      <div>Main Photo</div>
+      <img src={mainPhoto} alt="mainPhoto" width="100%" height="200" />
+      <input
+        type="file"
+        ref={hiddenMainPhotoInput}
+        accept="image/*"
+        style={{ display: 'none' }}
+        onChange={previewMainPhoto}
+      />
```

Comment:
In your next refactor, you may need to adjust the img attributes in order for the main photo to keep its aspect ratio but be clipped to fill the desired area for all screen sizes.